src/ray/raylet/node_manager.cc

```diff
@@ -369,66 +371,23 @@ void NodeManager::ClientAdded(const ClientTableDataT &client_data) {
     return;
   }
 
-  // TODO(atumanov): make remote client lookup O(1)
-  if (std::find(remote_clients_.begin(), remote_clients_.end(), client_id) ==
-      remote_clients_.end()) {
-    remote_clients_.push_back(client_id);
-  } else {
-    // NodeManager connection to this client was already established.
-    RAY_LOG(DEBUG) << "Received a new client connection that already exists: "
+  auto entry = node_manager_clients_.find(client_id);
```

Comment:
would `remote_node_managers_` be more explicit here?

Reply:
Considering the class name is `NodeManagerClient`, I think this name is fine. I was also thinking of naming the client/server as `NodeManagerRpcClient/NodeManagerRpcServer`. But not sure if that's too verbose.

Reply:
I was thinking it's probably not quite intuitive when people see `node_manager_clients_` in `NodeManager` code, `remote_node_managers_` indicates its meaning more directly. But I don't feel strong on this, it's ok to keep it as is.

Reply:
makes sense. I'll rename it to `remote_node_manager_clients_`